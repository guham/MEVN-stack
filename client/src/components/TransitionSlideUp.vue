<template>
  <Transition
    :css="false"
    @before-enter="beforeEnter"
    @enter="enter"
    @leave="leave"
  >
    <slot />
  </Transition>
</template>

<script>
import anime from 'animejs';

export default {
  name: 'TransitionSlideUp',
  props: {
    duration: {
      type: Number,
      required: true,
    },
  },
  methods: {
    beforeEnter(el) {
      el.style.transform = 'translateY(100%)';
    },
    enter(el, done) {
      anime({
        targets: el,
        translateY: 0,
        duration: this.duration,
        easing: 'easeOutQuad',
        complete: done,
      });
    },
    leave(el, done) {
      anime({
        targets: el,
        translateY: '100%',
        duration: this.duration,
        easing: 'easeInQuad',
        complete: done,
      });
    },
  },
};
</script>
