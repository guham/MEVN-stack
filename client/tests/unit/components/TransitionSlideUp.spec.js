import TransitionSlideUp from '@/components/TransitionSlideUp.vue';
import factory from '../factory';

describe('TransitionSlideUp.vue', () => {
  test('has the expected html structure', () => {
    const duration = 300;
    const wrapper = factory(TransitionSlideUp, {}, { duration });
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });
});
