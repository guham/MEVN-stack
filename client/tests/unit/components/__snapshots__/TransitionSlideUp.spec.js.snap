// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransitionSlideUp.vue has the expected html structure 1`] = `undefined`;
